src/Categories/Category/Monoidal/Instance/Rels.agda

```diff
@@ -1,204 +1,284 @@
-{-# OPTIONS --without-K --safe #-}
-module Categories.Category.Monoidal.Instance.Rels where
-
--- The category of relations is cartesian and (by self-duality) co-cartesian.
--- Perhaps slightly counter-intuitively if you're used to categories which act
--- like Sets, the product acts on objects as the disjoint union.
--- It also exhibits another monoidality, based on cartesian products of sets,
--- which is symmetric and closed.
+{-# OPTIONS --safe --without-K #-}
 
-open import Data.Empty.Polymorphic using (⊥; ⊥-elim)
-import Data.Product as ×
-open × using (_,_)
-open import Data.Sum using (_⊎_; inj₁; inj₂; [_,_]′)
-open import Data.Unit.Polymorphic using (⊤)
-open import Function using (case_of_; flip)
-open import Level using (Lift; lift)
-open import Relation.Binary.PropositionalEquality using (_≡_; refl)
+module Categories.Category.Monoidal.Instance.Rels where
 
 open import Categories.Category.Cartesian using (Cartesian)
-open import Categories.Category.Cartesian.Monoidal using (module CartesianMonoidal)
-open import Categories.Category.Core using (Category)
-open import Categories.Category.Cocartesian using (Cocartesian)
-open import Categories.Category.Instance.Rels using (Rels)
-open import Categories.Category.Monoidal using (Monoidal; monoidalHelper)
+open import Categories.Category.Instance.Rels
+open import Categories.Category.Monoidal.Core
 open import Categories.Category.Monoidal.Symmetric using (Symmetric; symmetricHelper)
 open import Categories.NaturalTransformation.NaturalIsomorphism using (niHelper)
 
+open import Data.Empty.Polymorphic using (⊥)
+open import Data.Product using (_×_; _,_; proj₁; proj₂)
+open import Data.Product.Relation.Binary.Pointwise.NonDependent using (×-setoid)
+open import Data.Sum.Base using (_⊎_; inj₁; inj₂; [_,_]′)
+open import Data.Sum.Relation.Binary.Pointwise using (⊎-setoid; inj₁; inj₂)
+open import Data.Unit.Polymorphic using (⊤)
+open import Function.Base using (case_of_)
+open import Level using (_⊔_; Lift; lift)
+open import Relation.Binary.Bundles using (Setoid)
+
 module _ {o ℓ} where
 
-  Rels-Cartesian : Cartesian (Rels o ℓ)
+  Rels-Cartesian : Cartesian (Rels o (o ⊔ ℓ))
   Rels-Cartesian = record
```

Comment:
So if anyone is going to use this record somewhere else, it is going to be a problem of sorts. Agda doesn't like huge explicit records. It much prefers records full of things that have local names, i.e. given using a `where` clause. That makes the unifier much happier.

Reply:
How huge is huge here? How much should I split the records up?

Reply:
It's not the size of the record, it's the size of the contents. 1-liners are fine, but embedded proofs with > 3 steps should definitely be named instead of inlined.